rust/RustLexer.g4

```diff
@@ -133,7 +133,9 @@ OUTER_BLOCK_DOC:
 
 BLOCK_COMMENT_OR_DOC: ( BLOCK_COMMENT | INNER_BLOCK_DOC | OUTER_BLOCK_DOC) -> channel (HIDDEN);
 
-SHEBANG: {this.SOF()}? '\ufeff'? '#!' ~[\r\n]* -> channel(HIDDEN);
+SHEBANG: {this.SOF()}? '\ufeff'? '#!' {this._input.LA(1) != 97}? ~[\r\n]* -> channel(HIDDEN);
```

Comment:
Strictly speaking, this is not "target agnostic" (https://github.com/antlr/grammars-v4/issues/4286#issuecomment-2427584163) because `!=` is a target-specific operator. The entire expression should go in a base class, which is already done for both [lexer](https://github.com/antlr/grammars-v4/blob/9f95ea41e82e347a841b01efa178605fc832925b/rust/RustLexer.g4#L29) and [parser](https://github.com/antlr/grammars-v4/blob/9f95ea41e82e347a841b01efa178605fc832925b/rust/RustParser.g4#L29). Please add a "is...()" method, reference that here, and implement the methods for each of the targets.